dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java

```diff
@@ -92,11 +94,26 @@ protected void executeImpl(CommandContext ctxt)  {
                     }
                 }
                 boolean doNormalSolrDocCleanUp = true;
+                RemoveLockAndUpdateDataset(ctxt);
                 ctxt.index().indexDataset(doomed, doNormalSolrDocCleanUp);
                 return;
             }
 
             throw new IllegalCommandException("Cannot delete a released version", this);
         }
     }
+
+    /**
+     * Removes potential lock, otherwise the dataset would be permanently locked for user without ability to edit it.
+     */
+    private void RemoveLockAndUpdateDataset(CommandContext ctxt) {
+        doomed.getLocks().stream()
+              .filter(datasetLock -> datasetLock.getReason().equals(DatasetLock.Reason.InReview))
+              .findAny()
+              .ifPresent(datasetLock -> {
+                  ctxt.engine().submit(new RemoveLockCommand(getRequest(), doomed, DatasetLock.Reason.InReview));
+                  doomed.setModificationTime(new Timestamp(new Date().getTime()));
```

Comment:
Why we modify `modificationTime`?

Reply:
Same thing is done in '~ReturnToAuthorCommand' so I wanted to stay consistent.

Reply:
Ok, let's leave it. As far as I can see 'ModificationTime' on 'dvobject' has effect on creating sitemap, and deciding if we should reindex - so it should be fine